File: c++/Reverse Words in a String.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <algorithm>

class Solution {
public:
    std::string reverseWords(std::string s) {
        std::vector<std::string> words;
        std::stringstream ss(s);
        std::string word;
        while (ss >> word) {
            words.push_back(word);
        }
        std::reverse(words.begin(), words.end());
        std::stringstream result;
        for (const std::string& w : words) {
            result << w << " ";
        }
        std::string reversed = result.str();
        return reversed.substr(0, reversed.size() - 1); 
    }
};

//test case
int main() {
    Solution sol;
    std::string s = "hello world";
    std::cout << sol.reverseWords(s) << std::endl;
    return 0;
}

File: python/Reverse Words in a String.py
class Solution:
    def reverseWords(self, s: str) -> str:
        words = s.split()
        words.reverse()
        return ' '.join(words)


def test_case():
    sol = Solution()
    assert sol.reverseWords("the sky is blue") == "blue is sky the"
    assert sol.reverseWords("  hello world  ") == "world hello"
    assert sol.reverseWords("a good   example") == "example good a"

    print("All test cases passed!")

test_case()
